### PR TITLE
Bug1626969: Expose `locale` through the RLB so that it can be set by consumers

### PR DESCRIPTION
**Pull request by travis79**
This exposes the `locale` from the internal `ClientInfoMetrics` in order to allow it to be set by consumers of the RLB.

### CODE REVIEW

File: docs/user/reference/metrics/test.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
+
```
Comment (badboy) [inline docs/user/reference/metrics/test.c:1]: ❓

File: glean-core/rlb/src/core_metrics.rs [hunk 2]
```diff
@@ -13,6 +13,9 @@ pub struct ClientInfoMetrics {
     pub app_display_version: String,
     /// The product-provided release channel (e.g. "beta").
     pub channel: Option<String>,
+    /// The locale of the application during initialization (e.g. "es-ES").
+    /// If the locale can't be determined on the system, the value is "und", to indicate "undetermined".
```
Comment (badboy) [inline glean-core/rlb/src/core_metrics.rs:17]: That doesn't apply here though, does it? If this field is unset we don't set it in the client info.
The foreign language bindings set it to `und`.

**Review by badboy — CHANGES_REQUESTED** [review]

File: docs/user/reference/metrics/test.c (hunk 1 above)
Comment (travis79) [reply docs/user/reference/metrics/test.c:1]: Must have had some random new file I opened in VSCode and accidentally committed it.

Comment (travis79) [reply docs/user/reference/metrics/test.c:1]: Oh, haha I know what it is...  it's a bitmap of a bird logo

**Comment by travis79** [comment]
Ugh, starting from scratch

**Pull request closed without merging**